azure-servicebus/src/main/java/com/microsoft/azure/servicebus/amqp/IAmqpConnection.java

```diff
@@ -4,11 +4,16 @@
  */
 package com.microsoft.azure.servicebus.amqp;
 
+import com.microsoft.azure.servicebus.ClientSettings;
 import org.apache.qpid.proton.amqp.transport.ErrorCondition;
 import org.apache.qpid.proton.engine.Link;
 
 public interface IAmqpConnection
 {
+	String getHostName();
+
+	ClientSettings getClientSettings();
```

Comment:
These two mehods are not necessary. Just check for type MesagingFactory in ConnectionHandler and use explicit type cast to get client settings from messaging factory.

Reply:
I have added a commit that should resolve this.